### PR TITLE
Use correct host name when writing ~/.netrc

### PR DESCRIPTION
**Pull request by BanzaiMan**
Use the hostname given in the payload

### CODE REVIEW

**Comment by BanzaiMan** [comment]
See https://github.com/travis-ci/travis-build/pull/804/files#r102742878 for motivation.

File: lib/travis/build/git/clone.rb [hunk 1]
```diff
@@ -88,14 +88,19 @@ def delete_netrc
           end
 
           def github?
+            source_host_name.downcase == 'github.com' || source_host_name.downcase.end_with?('.github.com')
```
Comment (svenfuchs) [inline lib/travis/build/git/clone.rb]: could this just be `source_host_name.downcase.end_with?('github.com')`? or `source_host_name.downcase =~ /\bgithub\.com/`

Comment (BanzaiMan) [inline lib/travis/build/git/clone.rb]: Neither seems to indicate the intent clear. `source_host_name.downcase.end_with?('github.com')` is incorrect, as `evilgithub.com` would match. I want the last 2 levels of `source_host_name` to match `github` and `com`, _*exactly*_.

Maybe `source_host_name.split('.')[-2,2] == ['github', 'com']`, but that seems rather unsightly.

Comment (svenfuchs) [inline lib/travis/build/git/clone.rb]: @BanzaiMan right, ignore me :)

**Pull request merged**